File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ChatCompletionsApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ChatCompletionsApi {
    private final HttpClient client;
    
    public ChatCompletionsApi(HttpClient client) {
        this.client = client;
    }

    /** Get chat completion */
    public ChatCompletionResponse getChatCompletion(String completionId) throws Exception {
        return (ChatCompletionResponse) client.get(ApiPaths.aiPath("/management/chat/completions/" + completionId + ""));
    }

    /** Update chat completion */
    public ChatCompletionResponse updateChatCompletion(String completionId, UpdateChatCompletionRequest body) throws Exception {
        return (ChatCompletionResponse) client.post(ApiPaths.aiPath("/management/chat/completions/" + completionId + ""), body);
    }

    /** Delete chat completion */
    public ChatCompletionDeleteResponse deleteChatCompletion(String completionId) throws Exception {
        return (ChatCompletionDeleteResponse) client.delete(ApiPaths.aiPath("/management/chat/completions/" + completionId + ""));
    }

    /** Get chat messages */
    public ChatMessageList getChatMessages(String completionId, Map<String, Object> params) throws Exception {
        return (ChatMessageList) client.get(ApiPaths.aiPath("/management/chat/completions/" + completionId + "/messages"), params);
    }

    /** List chat completions */
    public ChatCompletionList listChatCompletions(Map<String, Object> params) throws Exception {
        return (ChatCompletionList) client.get(ApiPaths.aiPath("/management/chat/completions"), params);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ChatCompletionsApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ChatCompletionsApi(private val client: HttpClient) {

    /** Get chat completion */
    suspend fun getChatCompletion(completionId: String): ChatCompletionResponse? {
        return client.get(ApiPaths.aiPath("/management/chat/completions/$completionId")) as? ChatCompletionResponse
    }

    /** Update chat completion */
    suspend fun updateChatCompletion(completionId: String, body: UpdateChatCompletionRequest): ChatCompletionResponse? {
        return client.post(ApiPaths.aiPath("/management/chat/completions/$completionId"), body) as? ChatCompletionResponse
    }

    /** Delete chat completion */
    suspend fun deleteChatCompletion(completionId: String): ChatCompletionDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/management/chat/completions/$completionId")) as? ChatCompletionDeleteResponse
    }

    /** Get chat messages */
    suspend fun getChatMessages(completionId: String, params: Map<String, Any>? = null): ChatMessageList? {
        return client.get(ApiPaths.aiPath("/management/chat/completions/$completionId/messages"), params) as? ChatMessageList
    }

    /** List chat completions */
    suspend fun listChatCompletions(params: Map<String, Any>? = null): ChatCompletionList? {
        return client.get(ApiPaths.aiPath("/management/chat/completions"), params) as? ChatCompletionList
    }
}
